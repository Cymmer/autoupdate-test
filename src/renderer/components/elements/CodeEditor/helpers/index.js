import { languages } from 'prismjs/components/prism-core';
import path from 'path';
import GLOBALS from 'codechum-app-globals';

export const convertLanguageExtensionToMonacoMode = (languageExtension) => {
  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.C) {
    return 'c';
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.CPP) {
    return 'cpp';
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.CSHARP) {
    return 'csharp';
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.JAVA) {
    return 'java';
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.PYTHON) {
    return 'python';
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.JAVASCRIPT) {
    return 'javascript';
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.ASSEMBLY) {
    return null;
  }

  return languageExtension;
};

export const convertLanguageExtensionToMobileEditorMode = (
  languageExtension
) => {
  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.C) {
    return languages.c;
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.CPP) {
    return languages.cpp;
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.CSHARP) {
    return languages.csharp;
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.JAVA) {
    return languages.java;
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.PYTHON) {
    return languages.python;
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.JAVASCRIPT) {
    return languages.js;
  }

  if (languageExtension === GLOBALS.LANGUAGE_EXTENSIONS.ASSEMBLY) {
    return languages.asm6502;
  }

  return languageExtension;
};

// utility functions to make the monaco editor react package work
// https://www.npmjs.com/package/@monaco-editor/react#for-electron-users

const ensureFirstBackSlash = (str) =>
  str.length > 0 && str.charAt(0) !== '/' ? `/${str}` : str;

export const uriFromPath = (_path) => {
  const pathName = path.resolve(_path).replace(/\\/g, '/');
  return encodeURI(`file://${ensureFirstBackSlash(pathName)}`);
};
