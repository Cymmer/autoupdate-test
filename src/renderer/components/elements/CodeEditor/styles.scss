@import '../../../scss/themes';
@import '../../../scss/mixins';
@import '../../../scss/variables';

/* here */
.MobileEditor {
  min-height: 420px;
  counter-reset: line;
  @include theme {
    background-color: t('editorBackgroundColor');
  }
  border-radius: 0 0 16px 16px;
}

.MobileEditor #codeArea {
  outline: none;
  padding-left: 48px !important;
}

.MobileEditor pre {
  padding-left: 48px !important;
}

.MobileEditor .editorLineNumber {
  position: absolute;
  left: 0;
  @include theme {
    color: t('textColor-300');
  }
  text-align: right;
  width: 32px;
  font-weight: 100;
}
