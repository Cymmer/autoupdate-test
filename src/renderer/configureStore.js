import { createHashHistory } from 'history';
import { routerMiddleware } from 'react-router-redux';
import { applyMiddleware, compose, createStore } from 'redux';
import { persistReducer } from 'redux-persist';
import hardSet from 'redux-persist/lib/stateReconciler/hardSet';
import storage from 'redux-persist/lib/storage';
import rootReducer from './ducks/index';

export const configureStore = () => {
  // create middlewares
  const history = createHashHistory();

  const composeEnhancers =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const middleware = applyMiddleware(routerMiddleware(history));

  // create a persist version of the reducer
  const persistConfig = {
    key: 'codechum-desktop',
    storage,
    stateReconciler: hardSet,
  };
  const persistedReducer = persistReducer(persistConfig, rootReducer);

  // create store
  const store = createStore(persistedReducer, composeEnhancers(middleware));

  return store;
};
