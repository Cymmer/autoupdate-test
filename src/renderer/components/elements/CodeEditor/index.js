/* eslint-disable @typescript-eslint/no-unused-vars */
import React, {
  forwardRef,
  useEffect,
  useState,
  useRef,
  useImperativeHandle,
} from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import path from 'path';
import './styles.scss';

import Editor, { loader } from '@monaco-editor/react';
import MobileEditor from 'react-simple-code-editor';
import { highlight } from 'prismjs/components/prism-core';
// eslint-disable-next-line no-unused-vars
import Prism from 'prismjs';
import 'prismjs/components/prism-clike';
import 'prismjs/components/prism-javascript';
import 'prismjs/components/prism-c';
import 'prismjs/components/prism-csharp';
import 'prismjs/components/prism-cpp';
import 'prismjs/components/prism-java';
import 'prismjs/components/prism-python';
import 'prismjs/components/prism-asm6502';
import 'prismjs/themes/prism-solarizedlight.css';

import { getIsNightMode } from 'ducks';

import GLOBALS from 'codechum-app-globals';
import { useWindowDimensions } from 'hooks';
import {
  convertLanguageExtensionToMonacoMode,
  convertLanguageExtensionToMobileEditorMode,
  uriFromPath,
} from './helpers';
import Spinner from '../Spinner';

// fix for exports is not defined error
// https://github.com/suren-atoyan/monaco-react/issues/87#issuecomment-942600539
// eslint-disable-next-line
self.module = undefined;

const CodeEditor = (
  {
    isNightMode,
    sourceCode,
    languageExtension,
    onCodeChange,
    editorOptionsOverride,
    readOnly,
    className,
    height,
  },
  ref
) => {
  const editorRef = useRef();
  const [codeValue, setCodeValue] = useState(sourceCode);

  useEffect(() => {
    console.log('__dirname', __dirname);
    console.log('process.env', process.env);
    
    // '../../../../../../../../node_modules/monaco-editor/min/vs'

    loader.config({
      paths: {
        vs: uriFromPath(
          path.join(__dirname, '../../../dependencies/monaco-editor/min/vs')
        ),
      },
    });
  }, []);

  const handleEditorDidMount = (editor) => {
    editor.focus();
    editor.setValue(sourceCode || '');

    editorRef.current = editor;
    editorRef.current.onDidChangeModelContent(() => {
      onCodeChange(editorRef.current.getValue());
      setCodeValue(editorRef.current.getValue());
    });
  };

  const EDITOR_OPTIONS = {
    readOnly,
    fontFamily: 'Monaco',
    fontSize: 14,
  };

  const mobileEditorRef = useRef(null);

  const highlightWithLineNumbers = (input, language) => {
    if (input && language) {
      return highlight(input, language)
        ?.split('\n')
        .map(
          (line, i) => `<span class='editorLineNumber'>${i + 1}</span>${line}`
        )
        .join('\n');
    }

    return null;
  };

  const { width: windowWidth } = useWindowDimensions();

  if (mobileEditorRef?.current?._input && readOnly) {
    // since the `react-simple-code-editor` library doesn't
    // expose a `disabled` or `readOnly` property, we manually
    // set its textarea to disabled by finding its `_input` element.
    //
    // NOTE: If the `react-simple-code-editor` library is updated,
    // this may need to be updated as well

    // eslint-disable-next-line
    ReactDOM.findDOMNode(mobileEditorRef?.current?._input).setAttribute(
      'disabled',
      true
    );
  }

  // Handles manual update of the code editor
  // this is usually called by parent component
  useImperativeHandle(
    ref,
    () => ({
      setValue: (value) => {
        // eslint-disable-next-line no-unused-expressions
        editorRef.current?.setValue(value);
        setCodeValue(value);
      },
    }),
    [editorRef]
  );

  return (
    <>
      {windowWidth > 767 ? (
        <Editor
          className={className}
          height={height}
          language={convertLanguageExtensionToMonacoMode(languageExtension)}
          loading={<Spinner />}
          options={{ ...EDITOR_OPTIONS, ...editorOptionsOverride }}
          theme={isNightMode ? 'dark' : 'light'}
          onMount={handleEditorDidMount}
        />
      ) : (
        <MobileEditor
          ref={mobileEditorRef}
          className="MobileEditor"
          data-test="mobileCodeEditor"
          highlight={(code) =>
            highlightWithLineNumbers(
              // so that there is an initial line of code, we should add an empty space
              // if there is no code yet. This space won't actually show in the code editor
              code || ' ',
              convertLanguageExtensionToMobileEditorMode(languageExtension)
            )
          }
          id="mobileCodeEditor"
          padding={12}
          style={{
            fontFamily: '"Monaco", monospace',
            fontSize: 14,
            height: '100%',
            outline: 0,
          }}
          textareaId="codeArea"
          value={codeValue}
          onValueChange={
            !readOnly
              ? (code) => {
                  setCodeValue(code);
                  onCodeChange(code);
                }
              : () => {}
          }
        />
      )}
    </>
  );
};

CodeEditor.defaultProps = {
  sourceCode: '',
  editorOptionsOverride: {},
  onCodeChange: null,
  readOnly: false,
  className: null,
  height: '100%',
};

CodeEditor.propTypes = {
  // the programming language set for this editor
  languageExtension: PropTypes.oneOf(GLOBALS.LANGUAGE_EXTENSIONS_PROP_TYPES)
    .isRequired,

  // the function that'll be called when the code is updated
  onCodeChange: PropTypes.func,

  // the source code currently displayed
  sourceCode: PropTypes.string,

  // true if the theme is night mode
  isNightMode: PropTypes.bool.isRequired,

  readOnly: PropTypes.bool,

  editorOptionsOverride: PropTypes.object,

  className: PropTypes.string,

  height: PropTypes.string,
};

const mapStateToProps = (store) => ({
  isNightMode: getIsNightMode(store),
});

export default connect(mapStateToProps, null, null, { forwardRef: true })(
  forwardRef(CodeEditor)
);
