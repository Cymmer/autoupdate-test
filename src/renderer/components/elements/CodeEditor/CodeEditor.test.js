import GLOBALS from 'codechum-app-globals';
import { mount } from 'enzyme';
import matchMediaPolyfill from 'mq-polyfill';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { configureStore } from '../../../configureStore';
import CodeEditor from './index';

beforeAll(() => {
  matchMediaPolyfill(window);
  window.resizeTo = function resizeTo(width, height) {
    Object.assign(this, {
      innerWidth: width,
      innerHeight: height,
      outerWidth: width,
      outerHeight: height,
    }).dispatchEvent(new this.Event('resize'));
  };
});

describe('RENDERING - CodeEditor', () => {
  it('should render with default props', () => {
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <CodeEditor languageExtension={GLOBALS.LANGUAGE_EXTENSIONS.C} />
      </Provider>
    );
    expect(component).toMatchSnapshot();

    component.unmount();
  });

  it('should render the normal editor on desktop browsers', () => {
    act(() => {
      window.resizeTo(768, 1000);
    });

    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <CodeEditor languageExtension={GLOBALS.LANGUAGE_EXTENSIONS.C} />
      </Provider>
    );
    expect(component).toMatchSnapshot();

    component.unmount();
  });

  it('should render the mobile editor on mobile devices (1)', async () => {
    await act(async () => {
      window.resizeTo(767, 1000);
    });

    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <CodeEditor languageExtension={GLOBALS.LANGUAGE_EXTENSIONS.C} />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });

  it('should render the mobile editor on mobile devices (2)', () => {
    act(() => {
      window.resizeTo(600, 1000);
      return undefined;
    });

    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <CodeEditor languageExtension={GLOBALS.LANGUAGE_EXTENSIONS.C} />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });

  // NOTE: Commented it out temporarily since need to investigate on why this test case is failing
  // eslint-disable-next-line jest/no-commented-out-tests
  // it('should pass the correct function for the `highlight` prop of the mobile editor', () => {
  //   act(() => {
  //     window.resizeTo(600, 1000);
  //     return undefined;
  //   });

  //   const component = mount(
  //     wrapWithStoreAndRouter(
  //       <CodeEditor languageExtension={GLOBALS.LANGUAGE_EXTENSIONS.PYTHON} />
  //     )
  //   );

  //   const mobileCodeEditor = findByTestAttr(
  //     component,
  //     'mobileCodeEditor'
  //   ).first();
  //   expect(
  //     typeof mobileCodeEditor.prop('highlight')(
  //       'print("Hello World")',
  //       GLOBALS.LANGUAGE_EXTENSIONS.PYTHON
  //     )
  //   ).toBe('string');
  // });
});
