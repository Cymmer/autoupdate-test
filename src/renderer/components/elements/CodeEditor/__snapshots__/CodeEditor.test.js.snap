// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - CodeEditor should render the mobile editor on mobile devices (1) 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Component)
    languageExtension="c"
  >
    <Connect(Component)
      languageExtension="c"
      reactReduxForwardedRef={null}
    >
      <ForwardRef(CodeEditor)
        dispatch={[Function]}
        isNightMode={false}
        languageExtension="c"
      >
        <Editor
          className="MobileEditor"
          data-test="mobileCodeEditor"
          highlight={[Function]}
          id="mobileCodeEditor"
          ignoreTabKey={false}
          insertSpaces={true}
          onValueChange={[Function]}
          padding={12}
          style={
            Object {
              "fontFamily": "\\"Monaco\\", monospace",
              "fontSize": 14,
              "height": "100%",
              "outline": 0,
            }
          }
          tabSize={2}
          textareaId="codeArea"
        >
          <div
            className="MobileEditor"
            data-test="mobileCodeEditor"
            id="mobileCodeEditor"
            style={
              Object {
                "boxSizing": "border-box",
                "fontFamily": "\\"Monaco\\", monospace",
                "fontSize": 14,
                "height": "100%",
                "outline": 0,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "textAlign": "left",
              }
            }
          >
            <textarea
              autoCapitalize="off"
              autoComplete="off"
              autoCorrect="off"
              className="npm__react-simple-code-editor__textarea"
              data-gramm={false}
              id="codeArea"
              onChange={[Function]}
              onKeyDown={[Function]}
              spellCheck={false}
              style={
                Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "WebkitTextFillColor": "transparent",
                  "background": "none",
                  "border": 0,
                  "boxSizing": "inherit",
                  "color": "inherit",
                  "display": "inherit",
                  "fontFamily": "inherit",
                  "fontSize": "inherit",
                  "fontStyle": "inherit",
                  "fontVariantLigatures": "inherit",
                  "fontWeight": "inherit",
                  "height": "100%",
                  "left": 0,
                  "letterSpacing": "inherit",
                  "lineHeight": "inherit",
                  "margin": 0,
                  "overflow": "hidden",
                  "overflowWrap": "break-word",
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                  "position": "absolute",
                  "resize": "none",
                  "tabSize": "inherit",
                  "textIndent": "inherit",
                  "textRendering": "inherit",
                  "textTransform": "inherit",
                  "top": 0,
                  "whiteSpace": "pre-wrap",
                  "width": "100%",
                  "wordBreak": "keep-all",
                }
              }
            />
            <pre
              aria-hidden="true"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "inherit",
                  "display": "inherit",
                  "fontFamily": "inherit",
                  "fontSize": "inherit",
                  "fontStyle": "inherit",
                  "fontVariantLigatures": "inherit",
                  "fontWeight": "inherit",
                  "letterSpacing": "inherit",
                  "lineHeight": "inherit",
                  "margin": 0,
                  "overflowWrap": "break-word",
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                  "pointerEvents": "none",
                  "position": "relative",
                  "tabSize": "inherit",
                  "textIndent": "inherit",
                  "textRendering": "inherit",
                  "textTransform": "inherit",
                  "whiteSpace": "pre-wrap",
                  "wordBreak": "keep-all",
                }
              }
            />
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": "
/**
 * Reset the text fill color so that placeholder is visible
 */
.npm__react-simple-code-editor__textarea:empty {
  -webkit-text-fill-color: inherit !important;
}

/**
 * Hack to apply on some CSS on IE10 and IE11
 */
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  /**
    * IE doesn't support '-webkit-text-fill-color'
    * So we use 'color: transparent' to make the text transparent on IE
    * Unlike other browsers, it doesn't affect caret color in IE
    */
  .npm__react-simple-code-editor__textarea {
    color: transparent !important;
  }

  .npm__react-simple-code-editor__textarea::selection {
    background-color: #accef7 !important;
    color: transparent !important;
  }
}
",
                }
              }
              type="text/css"
            />
          </div>
        </Editor>
      </ForwardRef(CodeEditor)>
    </Connect(Component)>
  </Connect(Component)>
</Provider>
`;

exports[`RENDERING - CodeEditor should render the mobile editor on mobile devices (2) 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Component)
    languageExtension="c"
  >
    <Connect(Component)
      languageExtension="c"
      reactReduxForwardedRef={null}
    >
      <ForwardRef(CodeEditor)
        dispatch={[Function]}
        isNightMode={false}
        languageExtension="c"
      >
        <Editor
          className="MobileEditor"
          data-test="mobileCodeEditor"
          highlight={[Function]}
          id="mobileCodeEditor"
          ignoreTabKey={false}
          insertSpaces={true}
          onValueChange={[Function]}
          padding={12}
          style={
            Object {
              "fontFamily": "\\"Monaco\\", monospace",
              "fontSize": 14,
              "height": "100%",
              "outline": 0,
            }
          }
          tabSize={2}
          textareaId="codeArea"
        >
          <div
            className="MobileEditor"
            data-test="mobileCodeEditor"
            id="mobileCodeEditor"
            style={
              Object {
                "boxSizing": "border-box",
                "fontFamily": "\\"Monaco\\", monospace",
                "fontSize": 14,
                "height": "100%",
                "outline": 0,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "textAlign": "left",
              }
            }
          >
            <textarea
              autoCapitalize="off"
              autoComplete="off"
              autoCorrect="off"
              className="npm__react-simple-code-editor__textarea"
              data-gramm={false}
              id="codeArea"
              onChange={[Function]}
              onKeyDown={[Function]}
              spellCheck={false}
              style={
                Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "WebkitTextFillColor": "transparent",
                  "background": "none",
                  "border": 0,
                  "boxSizing": "inherit",
                  "color": "inherit",
                  "display": "inherit",
                  "fontFamily": "inherit",
                  "fontSize": "inherit",
                  "fontStyle": "inherit",
                  "fontVariantLigatures": "inherit",
                  "fontWeight": "inherit",
                  "height": "100%",
                  "left": 0,
                  "letterSpacing": "inherit",
                  "lineHeight": "inherit",
                  "margin": 0,
                  "overflow": "hidden",
                  "overflowWrap": "break-word",
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                  "position": "absolute",
                  "resize": "none",
                  "tabSize": "inherit",
                  "textIndent": "inherit",
                  "textRendering": "inherit",
                  "textTransform": "inherit",
                  "top": 0,
                  "whiteSpace": "pre-wrap",
                  "width": "100%",
                  "wordBreak": "keep-all",
                }
              }
            />
            <pre
              aria-hidden="true"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "inherit",
                  "display": "inherit",
                  "fontFamily": "inherit",
                  "fontSize": "inherit",
                  "fontStyle": "inherit",
                  "fontVariantLigatures": "inherit",
                  "fontWeight": "inherit",
                  "letterSpacing": "inherit",
                  "lineHeight": "inherit",
                  "margin": 0,
                  "overflowWrap": "break-word",
                  "paddingBottom": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 12,
                  "pointerEvents": "none",
                  "position": "relative",
                  "tabSize": "inherit",
                  "textIndent": "inherit",
                  "textRendering": "inherit",
                  "textTransform": "inherit",
                  "whiteSpace": "pre-wrap",
                  "wordBreak": "keep-all",
                }
              }
            />
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": "
/**
 * Reset the text fill color so that placeholder is visible
 */
.npm__react-simple-code-editor__textarea:empty {
  -webkit-text-fill-color: inherit !important;
}

/**
 * Hack to apply on some CSS on IE10 and IE11
 */
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  /**
    * IE doesn't support '-webkit-text-fill-color'
    * So we use 'color: transparent' to make the text transparent on IE
    * Unlike other browsers, it doesn't affect caret color in IE
    */
  .npm__react-simple-code-editor__textarea {
    color: transparent !important;
  }

  .npm__react-simple-code-editor__textarea::selection {
    background-color: #accef7 !important;
    color: transparent !important;
  }
}
",
                }
              }
              type="text/css"
            />
          </div>
        </Editor>
      </ForwardRef(CodeEditor)>
    </Connect(Component)>
  </Connect(Component)>
</Provider>
`;

exports[`RENDERING - CodeEditor should render the normal editor on desktop browsers 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Component)
    languageExtension="c"
  >
    <Connect(Component)
      languageExtension="c"
      reactReduxForwardedRef={null}
    >
      <ForwardRef(CodeEditor)
        dispatch={[Function]}
        isNightMode={false}
        languageExtension="c"
      >
        <Editor
          language="c"
          loading={
            <Spinner
              className={null}
              colorName="blue"
              size="lg"
            />
          }
          onMount={[Function]}
          options={
            Object {
              "fontFamily": "Monaco",
              "fontSize": 14,
              "readOnly": undefined,
            }
          }
          theme="light"
        >
          <Memo(MonacoContainer)
            _ref={
              Object {
                "current": <div
                  style="width: 100%; display: none;"
                />,
              }
            }
            height="100%"
            isEditorReady={false}
            loading={
              <Spinner
                className={null}
                colorName="blue"
                size="lg"
              />
            }
            width="100%"
            wrapperProps={Object {}}
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": "100%",
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <Loading
                content={
                  <Spinner
                    className={null}
                    colorName="blue"
                    size="lg"
                  />
                }
              >
                <div
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "100%",
                    }
                  }
                >
                  <Spinner
                    className={null}
                    colorName="blue"
                    size="lg"
                  >
                    <div
                      className="Spinner_container"
                    >
                      <svg
                        className="Spinner___lg"
                        viewBox="25 25 50 50"
                      >
                        <circle
                          className="Spinner_circle___blue"
                          cx="50"
                          cy="50"
                          r="20"
                        />
                      </svg>
                    </div>
                  </Spinner>
                </div>
              </Loading>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </Memo(MonacoContainer)>
        </Editor>
      </ForwardRef(CodeEditor)>
    </Connect(Component)>
  </Connect(Component)>
</Provider>
`;

exports[`RENDERING - CodeEditor should render with default props 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Component)
    languageExtension="c"
  >
    <Connect(Component)
      languageExtension="c"
      reactReduxForwardedRef={null}
    >
      <ForwardRef(CodeEditor)
        dispatch={[Function]}
        isNightMode={false}
        languageExtension="c"
      >
        <Editor
          language="c"
          loading={
            <Spinner
              className={null}
              colorName="blue"
              size="lg"
            />
          }
          onMount={[Function]}
          options={
            Object {
              "fontFamily": "Monaco",
              "fontSize": 14,
              "readOnly": undefined,
            }
          }
          theme="light"
        >
          <Memo(MonacoContainer)
            _ref={
              Object {
                "current": <div
                  style="width: 100%; display: none;"
                />,
              }
            }
            height="100%"
            isEditorReady={false}
            loading={
              <Spinner
                className={null}
                colorName="blue"
                size="lg"
              />
            }
            width="100%"
            wrapperProps={Object {}}
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": "100%",
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <Loading
                content={
                  <Spinner
                    className={null}
                    colorName="blue"
                    size="lg"
                  />
                }
              >
                <div
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "100%",
                    }
                  }
                >
                  <Spinner
                    className={null}
                    colorName="blue"
                    size="lg"
                  >
                    <div
                      className="Spinner_container"
                    >
                      <svg
                        className="Spinner___lg"
                        viewBox="25 25 50 50"
                      >
                        <circle
                          className="Spinner_circle___blue"
                          cx="50"
                          cy="50"
                          r="20"
                        />
                      </svg>
                    </div>
                  </Spinner>
                </div>
              </Loading>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </Memo(MonacoContainer)>
        </Editor>
      </ForwardRef(CodeEditor)>
    </Connect(Component)>
  </Connect(Component)>
</Provider>
`;
